Inferences/ForwardDemodulation.cpp

```diff
@@ -167,9 +163,26 @@ bool ForwardDemodulationImpl<combinatorySupSupport>::perform(Clause* cl, Clause*
           }
         }
   #endif
-        if(!preordered && (_preorderedOnly || ordering.compare(trm,rhsS)!=Ordering::GREATER) ) {
+
+        auto subs = qr.unifier;
+        ASS(subs->isIdentityOnQueryWhenResultBound());
+
+        ScopedPtr<SubstApplicator> appl;
```

Comment:
Does this need to be a heap allocation at all, actually? I could imagine e.g.

```c++
Applicator<true> varSubst;
Applicator<false> notVarSubst;
SubstApplicator *appl = lhs.isVar() ? &varSubst : &notVarSubst;
```

(in which case, disregard previous comment about saving bytes)